homeassistant/components/hassio/handler.py

```diff
@@ -46,10 +46,14 @@ def api_data(funct):
 
     async def _wrapper(*argv, **kwargs):
         """Wrap function."""
-        data = await funct(*argv, **kwargs)
-        if data["result"] == "ok":
-            return data["data"]
-        raise HassioAPIError(data["message"])
+        while True:
```

Comment:
This shouldn't retry forever. It needs to fail eventually

Reply:
ACK